bugfix: Update stream-setting checkboxes from subscription events.

zee-bit opened the pull request:
This PR fixes the mismatch between stream settings if the `StreamInfo` popup is opened in ZT and subscription events are received from the server. The current approach registers a callback when the `StreamInfoView` is opened and looks for `subscription` events to trigger the callback, which in turn updates the checkbox UI.

Fixes #747.

zee-bit committed:
views: Add async function to update StreamInfo checkboxes from events.

This commit adds a function in StreamInfoView that runs asynchronously
at regular interval checking and updating the checkboxes. The function
is not yet executed from the StreamInfo view, because it doesnt yet
have the logic to stop the asynch function.

zee-bit commented:
@zulipbot add "feedback wanted"
@zulipbot add "further discussion required"

neiljp commented:
I'm wary from the description that a polling approach is the way to go with this, but rather that we do something like register a callback when opening the popup, which is triggered on events. That could be the starting point for a shift in how we glue parts of the application together, including loosening the coupling between the model and UI.

zee-bit committed:
model/views: Support event-triggered callback from views.

This commit adds two new functions in model.py:
- register_callback(): That registers callbacks for UI
elements and stores them in model.callbacks data-structure.
- process_callback(): That processes the corresponding callback
that are already registered after getting triggered from events.

These combined allow us to dynamically update the checkboxes in
stream_info popup after receiving events.

Fixes #747.

zee-bit commented:
I have already changed my approach from polling to callback-based. This works well as a v1, but I need some feedback on improvement so that I can generalize this for other UI elements that depend on events for updation.
- The first commit just adds an `asynch` function that will be responsible for updating the checkbox UI and will be passed as callback.
- The second commit defines the logic for registering and processing callbacks from `model` that will eventually be used to update the UI.
